adala/runtimes/base.py

```diff
@@ -15,10 +19,13 @@ class Runtime(BaseModelInRegistry):
     Attributes:
         verbose (bool): Flag indicating if runtime outputs should be verbose. Defaults to False.
         batch_size (Optional[int]): The batch size to use for processing records. Defaults to None.
+        concurrency (Optional[int]): The number of parallel processes to use for processing records. Defaults to -1 (utilize as many CPUs as available).
+                                    Note that when parallel processing is used, the memory footprint will be doubled compared to sequential processing.
     """
 
     verbose: bool = False
     batch_size: Optional[int] = None
+    concurrency: Optional[int] = Field(default=-1, alias='concurrent_clients')
```

Comment:
Maybe a nitpick, but calling this var "concurrency" for both  async and sync runtimes is possibly confusing, for async runtimes this is the level of concurrency and for sync runtimes it's the level of parallelism. Technically what we usually call parallelism in python is "concurrency plus parallelism", while what we call concurrency is "concurrency without parallelism". I can't think of a better name though, I guess it's okay :slightly_smiling_face:

Reply:
yeah, I'd agree with your descriptions, and while I agree that we typically mean something more specific when we say "concurrency", imo it's fair enough to use "concurrency" as the umbrella term (since it technically is, and I also can't think of a better umbrella term 🙂 )
Nikolai, if you have a better term up your sleeve, fantastic, but I'm personally good with "concurrency" too, fwiw